lstchain/io/io.py

```diff
@@ -1188,16 +1223,38 @@ def check_mc_type(filename):
     max_viewcone = simu_info.max_viewcone_radius.value
 
     if max_viewcone == 0.0:
-        mc_type = 'point_like'
+        mc_type = "point_like"
 
     elif min_viewcone == 0.0:
-        mc_type = 'diffuse'
+        mc_type = "diffuse"
 
     elif (max_viewcone - min_viewcone) < 0.1:
-        mc_type = 'ring_wobble'
+        mc_type = "ring_wobble"
 
     else:
-        raise ValueError('mc type cannot be identified')
+        raise ValueError("mc type cannot be identified")
 
     return mc_type
-            
+
+
+def get_mc_fov_offset(filename):
+    """
+    Check MC type ('point_like', 'diffuse', 'ring_wobble') based on the viewcone setting
+    Parameters
+    ----------
+    filename:path (DL1/DL2 hdf file)
+    Returns
+    -------
+    string
+    """
+
+    simu_info = read_simu_info_merged_hdf5(filename)
+
+    # Make sure we have full precision here
+    min_viewcone = simu_info.min_viewcone_radius.value.astype(float)
+    max_viewcone = simu_info.max_viewcone_radius.value.astype(float)
+
+    # This calculation is slightly more stable
+    mean_offset = 0.5 * min_viewcone - 0.5 * (max_viewcone - min_viewcone)
```

Comment:
Rather than calling it `mean_offset`, better to just call it `fov_offset`. 
Have you checked for diffuse MC files (not used for IRFs), as to what values you get for this parameter?

Reply:
I just checked one file and apparently the maximum is 6 deg. The minimum is 0 deg, so that would be 3 deg. How does that help? :sweat_smile: It is only calculated for point like or ring-wobble files.

Reply:
Not exactly useful here currently, but as the function is in lstchain.io.io, just wanted to be sure that this is applicable for all possible inputs here.

Reply:
And speaking of general inputs, we do have diffuse gamma MC for some special studies and for that case, the FOV offset evaluation as a mean value might not make much sense. Maybe there should be dependence with the check_mc_type function for this evaluation.

Reply:
Does it even make sense to compute an offset for diffuse monte carlo at all?
Like what do you even expect to be the result?

Reply:
Indeed, I dont think it is a usable parameter for diffuse MC, but we should have the MC type used for the creation of the IRFs, continued as a general metadata information. Maybe once the CTA IRF/DL3 Model requirement is complete, and we follow it exactly, then we will have a different way of saving such a metadata. I dont know if ctapipe has this right now (and in case it is available in > v0.12, then we can push these changes for ctapipe_0.17 branch instead), and if it is we might have to adapt the code accordingly.

In any case, having the G_OFFSET value as 6 deg does indicate that the simulation used is of diffuse type, and covering the full FoV.

Reply:
@maxnoe what do you suggest for the metadata `G_OFFSET` for diffuse MC-based IRF? Or in general, how should this information be saved as, as per CTAO?

Reply:
Sounds good! It will be clearer when reading the IRFs! 

Then, we would just have to update the same in https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/high_level/interpolate.py#L226, https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/high_level/interpolate.py#L429 and https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/tools/lstchain_create_dl3_file.py#L505-L510. @LukasNickel Can you try some options for these names?

Reply:
Proposal:
We get rid of it altogether.

For interpolation you need to know if the axes are compatible and you check that. Why would the axes be compatible if the IRF were computed from different offsets? If you insist, that the MC type needs to be known, we can just put that into the header directly instead of G_OFFSET. But I would argue that is not even necessary: If different simulation types / offsets produce the same irf binning, they are either compatible or incorrect.

Reply:
The logic seems correct to me. Can you test with the test sample MC (diffuse) and any particular node of a recent MC production (ring-wobble) to be certain of this?

I would still prefer to have readable metadata information telling the MC type used for a given IRF production. It will help analyzers be clear of the IRFs they work with, more easily than accessing logging data at least.

Reply:
I dont think analyzers should have to deal with checking the metadata of their IRFs, no?